components/identity-mgt/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/mail/DefaultEmailSendingModule.java

```diff
@@ -73,9 +73,20 @@ public void sendEmail() {
 
         try {
             if (this.notification == null) {
-                throw new NullPointerException("Notification not set. Please set the notification before sending messages");
+                throw new IllegalStateException("Notification not set. " +
+                        "Please set the notification before sending messages");
             }
             PrivilegedCarbonContext.startTenantFlow();
+            if (notificationData != null) {
+                String tenantDomain = notificationData.getDomainName();
+                PrivilegedCarbonContext carbonContext = PrivilegedCarbonContext.getThreadLocalCarbonContext();
+                carbonContext.setTenantDomain(tenantDomain);
+                carbonContext.getTenantId(true);
+            } else {
+                if (log.isDebugEnabled()) {
+                    log.debug("notification data not found. Tenant might not be loaded correctly");
```

Comment:
When notificationData is null, there's no method to get the details about tenant. Therefore no context details are added in the debug log.